src/data/aggregations/decisionDesk/getAllCongressData.ts

```diff
@@ -0,0 +1,97 @@
+import * as Sentry from '@sentry/node'
+
+import {
+  CandidatesWithVote,
+  CongressDataResponse,
+  GetAllCongressDataProps,
+  GetAllCongressDataResponse,
+  RacesVotingDataResponse,
+} from '@/data/aggregations/decisionDesk/types'
+import { getPoliticianFindMatch } from '@/data/aggregations/decisionDesk/utils'
+import { DTSI_AllPeopleQuery } from '@/data/dtsi/generated'
+import { queryDTSIAllPeople } from '@/data/dtsi/queries/queryDTSIAllPeople'
+import { getLogger } from '@/utils/shared/logger'
+
+const logger = getLogger('aggregations/decisionDesk/getAllCongressData')
+
+const reduceCongressData = (congressData: RacesVotingDataResponse[]) =>
+  congressData.reduce(
+    (acc, currentData) => ({
+      office: currentData.office,
+      year: currentData.year,
+      candidatesWithVotes: [
+        ...acc.candidatesWithVotes,
+        ...currentData.candidatesWithVotes.reduce((innerAcc, currentCandidate) => {
+          const existingCandidate = innerAcc.find(
+            currentInnerAcc => currentInnerAcc.id === currentCandidate.id,
+          )
+
+          return existingCandidate
+            ? innerAcc.map(currentInnerAcc =>
+                currentInnerAcc.id === currentCandidate.id
+                  ? { ...currentInnerAcc, votes: currentInnerAcc.votes + currentCandidate.votes }
+                  : currentInnerAcc,
+              )
+            : [...innerAcc, { ...currentCandidate }]
+        }, [] as CandidatesWithVote[]),
+      ],
+    }),
+    {
+      year: 0,
+      office: null,
+      candidatesWithVotes: [],
+    } as CongressDataResponse,
+  )
```

Comment:
Are you rewriting the acc `office` and `year` for every entry and in the end getting only one `office` and `year`? If that is the case would be easier to flatmap the `candidatesWithVote`  and map through everything to get the `candidatesWithVote` instead.

Something like this maybe? Did not test it, so may need some changes.
```suggestion
const reduceCongressData = (congressData: RacesVotingDataResponse[]) => {
    const candidatesWithVoteFlatMap = congressData.flatMap(currentData => currentData.candidatesWithVotes);

    return {
      office: congressData[0].office,
      year: congressData[0].year,
      candidatesWithVotes: candidatesWithVoteFlatMap.map(currentCandidate => {
        const candidateVotes = candidatesWithVoteFlatMap.filter(candidate => candidate.id === currentCandidate.id)
        const totalVotes = candidateVotes.reduce((acc, current) => acc + current.votes, 0)

        return {
          ...currentCandidate,
          votes: totalVotes,
      }}),
    }
  }
```

Reply:
you're definitely right, man, commiting your change thanks